classifier/__init__.py

```diff
@@ -1323,6 +1325,10 @@ def audience(self, genres=[], default_audience=None):
             audience = Classifier.AUDIENCE_YOUNG_ADULT
         elif total_adult_weight > 0:
             audience = Classifier.AUDIENCE_ADULT
+        elif all_ages_weight > 0:
+            audience = Classifier.AUDIENCE_ALL_AGES
+        elif research_weight > 0:
+            audience = Classifier.AUDIENCE_RESEARCH
```

Comment:
@leonardr I had to add these in order for the work to successfully save the value that was being saved in the admin. I'm actually not too sure if the logic is correct.

Reply:
This does need some work but I'm not 100% sure what the right answer is. Here's my best guess:

* Theoretically, all_ages_weight should be added to both total_adult_weight and total_juvenile_weight, but that cancels out when they're being compared directly, and its detrimental when we're making other comparisons, so leave it alone.
* If research_weight is larger than _either_ total_adult_weight + all_ages_weight _or_ total_juvenile_weight + all_ages_weight, then the audience is 'Research'
* If all_ages_weight is larger than _either_ total_adult_weight _or_ total_juvenile_weight, then the audience is 'All Ages'

Amy, does this sound right?

Reply:
I agree with the first point, it doesn't make sense to add the all ages weight to other weights if all ages itself is a possible output.

I'm not sure about the second point - if we have some evidence that a book is for YA or Adult but more evidence that it's for research, calling it research means we'll exclude it from other lanes/searches even though it might be of general interest. Maybe we'd rather do that than include things that might not be interesting?

For the rest, I agree but we might be missing an audience for YA books that are also interesting for adults - maybe that's most YA books though.

Reply:
@aslagle

Reply:
For the second point, it makes sense to not include something that may not be interesting.

Reply:
I'm thinking specifically of children's books from before... Dr. Seuss, basically. There are a ton of these and there's very strong bibliographic evidence for filing them in the children's section, but apart from a few classics nobody actually does. They're kept in research collections.

There are YA books of interest to adults, but at this point the interested adults know to go find them in the YA section of the library/bookstore. I don't even know what term we'd use to describe those audiences; everything I can think of starts with "YA, but..."

Reply:
I updated this to include the logic for the second point.